### PR TITLE
Handle zero-divisor in `_consumeClaimable*` after partial-claim rounding

### PR DESCRIPTION
**Pull request by Amxx**
Following https://audits.openzeppelin.com/openzeppelin-solidity/project/openzeppelin-community-contracts/41249254-ce3c-4bfa-8684-4c7863c775e7/issue/rounding-in-consumeclaimablewithdraw-can-exhaust-claimable-shares-before-assets-skewing-totalsupply-and-erc-4626-conversions

## Summary by CodeRabbit

* **Refactor**
  * Optimized deposit and redemption processing to improve performance through streamlined calculations for exact match scenarios.

* **Tests**
  * Added test coverage for rounding edge cases in deposit and redeem flows near conversion limits.

### CODE REVIEW

File: test/token/ERC20/extensions/ERC7540Admin.test.js [hunk 1]
```diff
@@ -15,7 +15,8 @@ const tokenName = 'Asset Token';
 const tokenSymbol = 'AST';
 
 describe('ERC7540Admin', function () {
-  for (const withTmpHolder of [false, true]) {
+  // for (const withTmpHolder of [false, true]) {
+  for (const withTmpHolder of [false]) {
```
Comment (ernestognw) [inline test/token/ERC20/extensions/ERC7540Admin.test.js]:
```suggestion
  for (const withTmpHolder of [false, true]) {
```

**Review by ernestognw — COMMENTED** [review]
LGTM, just a small detail on the tests

**Review by ernestognw — APPROVED** [review]
